fmpz_sparse_mat.h

```diff
@@ -0,0 +1,792 @@
+void fmpz_sparse_mat_solve_bound(fmpz_t N, fmpz_t D, const fmpz_sparse_mat_t A, const fmpz_mat_t B);
+
+FLINT_DLL
+int fmpz_sparse_mat_solve_fmpz_mat_dixon(fmpz_mat_t X, fmpz_t mod, const fmpz_sparse_mat_t A, const fmpz_mat_t B);
+
+FMPZ_SPARSE_MAT_INLINE
+int fmpz_sparse_mat_solve_fmpz_vec_dixon(fmpz * x, fmpz_t mod, const fmpz_sparse_mat_t A, fmpz *b)
+{
+    int ret;
+    slong i;
+    fmpz_mat_t X, B;
+    fmpz_mat_init(X, A->c, 1);
+    fmpz_mat_init(B, A->r, 1);
+    for (i = 0; i < A->r; ++i) fmpz_set(fmpz_mat_entry(B, i, 0), &b[i]);
+    ret = fmpz_sparse_mat_solve_fmpz_mat_dixon(X, mod, A, B);
+    for (i = 0; i < A->c; ++i) fmpz_set(&x[i], fmpz_mat_entry(X, i, 0));
+    fmpz_mat_clear(X);
+    fmpz_mat_clear(B);
+    return ret;
+}
+
+FLINT_DLL
+int fmpz_sparse_mat_solve_fmpz_mat_dixon_den(fmpz_mat_t X, fmpz_t den, const fmpz_sparse_mat_t A, const fmpz_mat_t B);
+
+FMPZ_SPARSE_MAT_INLINE
+int fmpz_sparse_mat_solve_fmpz_vec_dixon_den(fmpz * x, fmpz_t den, const fmpz_sparse_mat_t A, fmpz *b)
+{
+    int ret;
+    slong i;
+    fmpz_mat_t X, B;
+    fmpz_mat_init(X, A->c, 1);
+    fmpz_mat_init(B, A->r, 1);
+    for (i = 0; i < A->r; ++i) fmpz_set(fmpz_mat_entry(B, i, 0), &b[i]);
+    ret = fmpz_sparse_mat_solve_fmpz_mat_dixon_den(X, den, A, B);
+    for (i = 0; i < A->c; ++i) fmpz_set(&x[i], fmpz_mat_entry(X, i, 0));
+    fmpz_mat_clear(X);
+    fmpz_mat_clear(B);
+    return ret;
+}
+
+/* Determinant computation */
+FLINT_DLL
+void fmpz_sparse_mat_det_bound(fmpz_t bound, const fmpz_sparse_mat_t A);
+FLINT_DLL
+void fmpz_sparse_mat_det_cofactor(fmpz_t det, const fmpz_sparse_mat_t M);
+FLINT_DLL
+void fmpz_sparse_mat_det_bareiss(fmpz_t det, const fmpz_sparse_mat_t M);
+FLINT_DLL
+void fmpz_sparse_mat_det_divisor(fmpz_t d, const fmpz_sparse_mat_t M);
+FLINT_DLL
+void fmpz_sparse_mat_det_modular_accelerated(fmpz_t det, const fmpz_sparse_mat_t A, int proved);
+FLINT_DLL
+void fmpz_sparse_mat_det_modular_given_divisor(fmpz_t det, const fmpz_sparse_mat_t A, const fmpz_t d, int proved);
+FLINT_DLL
+void fmpz_sparse_mat_det_modular(fmpz_t det, const fmpz_sparse_mat_t A, int proved);
+FLINT_DLL
+void fmpz_sparse_mat_det(fmpz_t det, const fmpz_sparse_mat_t A);
+
+/* Fraction-free LU factorization */
+FLINT_DLL
+slong fmpz_sparse_mat_fflu(fmpz *D, slong *P, slong *Q, fmpz_sparse_mat_t L, fmpz_sparse_mat_t U, 
+                            const fmpz_sparse_mat_t M);
+
+/* Hermite normal form */
+FLINT_DLL
+int fmpz_sparse_mat_is_in_hnf(const fmpz_sparse_mat_t A);
+
+FLINT_DLL
+slong fmpz_sparse_mat_hnf_classical(fmpz_sparse_mat_t M);
+
+FLINT_DLL
+slong fmpz_sparse_mat_hnf_xgcd(fmpz_sparse_mat_t M);
+
+FLINT_DLL
+slong fmpz_sparse_mat_hnf_minors(fmpz_sparse_mat_t M);
+
+FLINT_DLL
+slong fmpz_sparse_mat_hnf_modular(fmpz_sparse_mat_t M, const fmpz_t det);
+
+FLINT_DLL
+slong fmpz_sparse_mat_hnf_modular_eldiv(fmpz_sparse_mat_t M, const fmpz_t n);
+
+/* Modular forms */
+slong fmpz_sparse_mat_howell_form_mod(fmpz_sparse_mat_t M, const fmpz_t mod);
```

Comment:
You need a FLINT_DLL here like the other functions.

Reply:
Ah, that will be it! I couldn't think of any other reason it would not find the function.